README.md

```diff
@@ -12,14 +12,36 @@ Low level access to Cortex-A processors.
 
 ## Minimum Supported Rust Version
 
-Requires rustc 1.45.0 or later due to use of the new `asm!()` syntax.
+Requires a recent nightly of Rust.
 
 ## Usage
 
-Example from https://github.com/rust-embedded/rust-raspi3-OS-tutorials
+Check out https://github.com/rust-embedded/rust-raspberrypi-OS-tutorials for usage examples.
+
+Please note that for using this crate's ARMv8 register definitions (as provided by
+`cortex_a::regs::*`), you need to also include
+[`tock-registers`](https://crates.io/crates/tock-registers) in your project. This is because the
+`interface` traits provided by `tock-registers` are used. You should include the same version of
+`tock-registers` as is being used by this crate to ensure sane interoperatbility.
+
+### Example
+
+```toml
+[dependencies]
+tock-registers = { version = "X.Y.Z", default-features = false, features = ["register_types"] }
```

Comment:
I think just specifying a version with the default feature set suffices for this example? Except if you explicitly do _not_ want `std_unit_tests` to be enabled because it fundamentally does not work with this crate or the ecosystem (for instance because of custom test runners).

In my opinion, Cargo crate features are a powerful, but sometimes also rather confusing tool, especially when crates interact with each other. I tried to make the `tock-registers` features as idiomatic as possible, while resorting to a default feature set which seems reasonable for most applications:
- `register_types` is enabled by default, as the included `ReadOnly`, `WriteOnly`, `ReadWrite`, `Aliased` and `InMemoryRegister` seem very useful for the overwhelming number cases where `tock-registers` is used. It doesn't hurt when it's enabled, feature unification does it's job just fine, but crates can easily choose to leave it disabled to not import any unsafe code: `#![cfg_attr(not(feature = "register_types"), forbid(unsafe_code))]`
- `std_unit_tests` is enabled by default, as the tests it performs are -- without a doubt -- very useful.
  
  I didn't test this, though my hypothesis is: if a crate has this feature enabled, and one of it's dependencies is incompatible with it (for example, through custom test frameworks), or macros cause incompatible code to be generated in the aforementioned crate, feature unification will likely cause test-compilation to error out in these cases, even though the library crate did not have this feature enabled.

  If this is indeed the case, it's a slight annoyance, but follows the fail-fast paradigm: rather than silently drop all tests as a disable-feature is propagated through feature unification (in retrospect this seemed kind of dangerous), now the test explicitly fail. This follows the fail-fast paradigm and especially when it comes to tests, is IMHO definitely preferred.

@andre-richter Do you have a different opinion on any of these points? I hope I've got this right, and if not would love to hear feedback. The currently `v0.7` release is great to these these sorts of things figured out before the upcoming `v1.0`.

Reply:
The reason why its not default feature set in this example is because I copied it over from `rust-raspberrypi-os-tutorials` where the subsequent Rust snippet is from as well.

But I agree that it might be confusing to see it with a modified feature set. I will decouple the `toml` and `rust` snippets when I get my next chance to work on it, and got for default feature set.

> I didn't test this, though my hypothesis is: if a crate has this feature enabled, and one of it's dependencies is incompatible with it (for example, through custom test frameworks), or macros cause incompatible code to be generated in the aforementioned crate, feature unification will likely cause test-compilation to error out

Compiler throws a "Can't find crate for test` error in this case. I'm not very happy with it, because it doesn't give away at first what happened. You need to enable more verbose backtracing to understand that `tock-registers` sneaked a `#[test]' into your code. But probably not much that can be done in this case. Anyways, not a deal breaker for now. Implementing `custom-test-frameworks` is still a major effort nowadays. If a user goes down this road, he will be able to eventually figure this one out.